### ISSUE TITLE
Issues with merged code

### ISSUE THREAD

**Issue by jkblc** [ubsuny/CP1-24-HW4#48]
There are some very frustrating errors in the already merged code and I'm not sure how to fully test my code without rewriting most of the existing codebase

**Comment by dnxjay** [comment]
are you able to reference the errors?

**Comment by jkblc** [comment]
File "/workspaces/CP1-24-HW4/Least-squares Fit/fit_and_plot.py", line 19, in <module>
    from linear_fit import linear_fit
  File "/workspaces/CP1-24-HW4/Least-squares Fit/linear_fit.py", line 79, in <module>
    x, y, sigma = read_data_from_csv(filename)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/CP1-24-HW4/Least-squares Fit/linear_fit.py", line 16, in read_data_from_csv
    df = pd.read_csv(filename)
         ^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1026, in read_csv
    return _read(filepath_or_buffer, kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 620, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1620, in __init__
    self._engine = self._make_engine(f, self.engine)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1880, in _make_engine
    self.handles = get_handle(
                   ^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/common.py", line 873, in get_handle
    handle = open(
             ^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'data.csv'

**Comment by jkblc** [comment]
This is part of the reason nobody's pylint is passing

**Comment by jkblc** [comment]
There is no central organization for files and directory name conventions are random

**Comment by iglesias-cardinale** [comment]
If not for this project then maybe for the next one, perhaps we should assign an organizer role. By this I mean someone who, near the end of the assignment, will do one final pull request that organizes all of the files/folders so that everything is operational and coherent.

This should obviously be in addition to having some sort of framework set at the beginning that everyone doing their PRs should try to fit within.

**Comment by jkblc** [comment]
I agree that it would be useful, but it would be good to do it after all pull requests have been merged or closed so we don't create a lot of conflicts for everyone to clean up

**Comment by jkblc** [comment]
Issues with organization will likely improve on an individual level as everyone learns conventions. I am just slow to change too much within my own pull request as we are limited to one

**Comment by iglesias-cardinale** [comment]
I agree. That's why I think it would be good for one person to do it all at the end. That way their organization can count as their one pull request for the assignment.

**Comment by jkblc** [comment]
Although The repository manager could be strict with PRs and enforce structure. And reviewers could actually check out and run PRs to make sure they actually work

**Comment by kylemasc917** [comment]
The problem I find is every assignment is different enough that the organization we come to use in one is not as useful as the next so fixing it after the fact could be a good fix and people should be more mindful that others may have to use their code.

**Comment by avgagliardo** [comment]
> The problem I find is every assignment is different enough that the organization we come to use in one is not as useful as the next so fixing it after the fact could be a good fix and people should be more mindful that others may have to use their code.

This is a super common problem to run into. I try my best to keep my own code modular and organized so that it can be easily organized after it's been merged. The maintainer (or someone similar) has a better vision of how the project should be organized so I try to make it simple for them.

**Comment by dnxjay** [comment]
> If not for this project then maybe for the next one, perhaps we should assign an organizer role. By this I mean someone who, near the end of the assignment, will do one final pull request that organizes all of the files/folders so that everything is operational and coherent.
> 
> This should obviously be in addition to having some sort of framework set at the beginning that everyone doing their PRs should try to fit within.

I think technically the person thats reviewing the PR are responsible for that, the only issue with that is that it can be subjective for each individual for maybe thats a good idea to have a designated person to do so.

**Comment by laserlab** [comment]
I will suggest a solution today in class for the next homework.

**Comment by dnxjay** [comment]
> Although The repository manager could be strict with PRs and enforce structure. And reviewers could actually check out and run PRs to make sure they actually work

i can agree with this

**Comment by haiderabbas007** [comment]
> File "/workspaces/CP1-24-HW4/Least-squares Fit/fit_and_plot.py", line 19, in from linear_fit import linear_fit File "/workspaces/CP1-24-HW4/Least-squares Fit/linear_fit.py", line 79, in x, y, sigma = read_data_from_csv(filename) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/workspaces/CP1-24-HW4/Least-squares Fit/linear_fit.py", line 16, in read_data_from_csv df = pd.read_csv(filename) ^^^^^^^^^^^^^^^^^^^^^ File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1026, in read_csv return _read(filepath_or_buffer, kwds) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 620, in _read parser = TextFileReader(filepath_or_buffer, **kwds) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1620, in **init** self._engine = self._make_engine(f, self.engine) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1880, in _make_engine self.handles = get_handle( ^^^^^^^^^^^ File "/home/codespace/.local/lib/python3.12/site-packages/pandas/io/common.py", line 873, in get_handle handle = open( ^^^^^ FileNotFoundError: [Errno 2] No such file or directory: 'data.csv'

I can explain the cause of this problem. The linear_fit.py file written by @dnxjay has an # Example Usage at the end 
![image](https://github.com/user-attachments/assets/0a768b9c-663e-4317-8bf1-95745d8a8a29)

Since it's just an example usage, of course there won't be any data.csv file or directory.
